From 0f2dba27dc8b706c5985ef05ed7acb0c7d15cf6b Mon Sep 17 00:00:00 2001
From: Jeremy Drake <github@jdrake.com>
Date: Sun, 13 Apr 2025 22:21:02 -0700
Subject: [PATCH 02/15] undef i386

i686 cygwin gcc has a default define of i386 to 1, which conflicts with
a namespace of i386, so undef it.
---
 llvm/include/llvm/ExecutionEngine/JITLink/i386.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/llvm/include/llvm/ExecutionEngine/JITLink/i386.h b/llvm/include/llvm/ExecutionEngine/JITLink/i386.h
index efe818293..6e53aae99 100644
--- a/llvm/include/llvm/ExecutionEngine/JITLink/i386.h
+++ b/llvm/include/llvm/ExecutionEngine/JITLink/i386.h
@@ -16,6 +16,8 @@
 #include "llvm/ExecutionEngine/JITLink/JITLink.h"
 #include "llvm/ExecutionEngine/JITLink/TableManager.h"
 
+#undef i386
+
 namespace llvm::jitlink::i386 {
 /// Represets i386 fixups
 enum EdgeKind_i386 : Edge::Kind {
-- 
2.49.0.windows.1

